sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/http/rest/RestProxyUtils.java

```diff
@@ -95,12 +96,73 @@ public static Context mergeRequestOptionsContext(Context context, RequestOptions
         Context optionsContext = options.getContext();
 
         if (optionsContext != null && optionsContext != Context.NONE) {
-            context = Context.mergeContexts(context, optionsContext);
+            context = mergeContexts(context, optionsContext);
         }
 
         return context;
     }
 
+    /**
+     * Merges data from two {@link Context} instances.
+     *
+     * @param into The {@link Context} to merge data into.
+     * @param from The {@link Context} to merge data from.
+     *
+     * @return The merged {@link Context}.
+     */
+    private static Context mergeContexts(Context into, Context from) {
+        Objects.requireNonNull(into, "'into' cannot be null.");
+        Objects.requireNonNull(from, "'from' cannot be null.");
```

Comment:
Given this is internal now, we could allow nullness here and just update the checks to

`into == null || into == Context.NONE -> from`
`from == null || from == Context.NONE -> into`